src/idpi/grib_decoder.py

```diff
@@ -82,6 +52,171 @@ def _extract_pv(pv):
     }
 
 
+@dc.dataclass
+class _FieldBuffer:
+    dims: tuple[str, ...] | None = None
+    hcoords: dict[str, xr.DataArray] = dc.field(default_factory=dict)
+    metadata: dict[str, typing.Any] = dc.field(default_factory=dict)
+    time_meta: dict[int, dict] = dc.field(default_factory=dict)
+    values: dict[tuple[int, ...], np.ndarray] = dc.field(default_factory=dict)
+
+    def load(self, field):
```

Comment:
This is a bit tricky because the actual object is defined in earthkit data. The main issue is that type hinting is not implemented in that library. I tried generating stubs but it was a difficult exercise due to reliance on duck typing and some inconsistencies between base and child classes. What I could do is define a [Protocol](https://mypy.readthedocs.io/en/stable/protocols.html#protocols-and-structural-subtyping) that describes the interfaces that are relied upon in this particular function.